helix-core/src/main/java/org/apache/helix/manager/zk/CurStateCarryOverUpdater.java

```diff
@@ -35,16 +35,17 @@
  */
 class CurStateCarryOverUpdater implements DataUpdater<ZNRecord> {
   final String _curSessionId;
-  final String _initState;
+  final Map<String, String> _initStateMap;
```

Comment:
You pass in the expected state map. But here you still name it as init state map, it could be confusing. Let's change the name.